napari/plugins/io.py

```diff
@@ -76,8 +83,20 @@ def read_data_with_plugins(
                 logger.error(err.format_with_contact_info())
 
 
-def write_data_with_plugins(
-    path: str, layer_data: LayerData, plugin_manager=None
+def write_layer_with_plugins(path: str, layer: Layer, plugin_manager=None):
+    # TODO: sidenote, I never liked this pattern, and it's getting less
+    # meaningful as we make more customizations of our plugin manager...
+    # but it might be necessary for tests?
+    plugin_manager = plugin_manager or napari_plugin_manager
+    data, meta, layer_type = layer.as_layer_tuple()
+    hook_caller = getattr(
+        napari_plugin_manager.hook, 'napari_write_' + layer_type
+    )
+    hook_caller(path=path, data=data, meta=meta)
+
+
+def write_layers_with_plugins(
```

Comment:
this is the main function, and internally calls `write_layer_with_plugins` if `len(layers) == 1`.  (we could make `_write_layer` private)